doc/source/cd/background.ipynb

```diff
@@ -0,0 +1,608 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Background\n",
+    "\n",
+    "* [0. What is drift?](#what_is_drift)\n",
+    "\n",
+    "* [1. Detecting drift](#drift_detection)\n",
+    "    * [Hypothesis testing](#hypothesis)\n",
+    "    * [Dimension reduction](#dimension_reduction)\n",
+    "    * [Simple example](#example)\n",
+    "  \n",
+    "* [2. Learned drift detection](#learned)\n",
+    "\n",
+    "* [3. Online detection](#online)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 0. What is drift? <a class=\"anchor\" id=\"what_is_drift\"></a>\n",
+    "\n",
+    "Although powerful, modern machine learning models can be sensitive. Seemingly subtle changes in a data distribution can destroy the performance of otherwise state-of-the art models, which can be especially problematic when ML models are deployed in production. Typically, ML models are tested on held out data in order to estimate their furture performance. Crucially, this assumes that the *process* underlying the input data $\\mathbf{X}$ and output data $\\mathbf{Y}$ remains constant. \n",
+    "\n",
+    "*Drift* is said to occur when the process underlying $\\mathbf{X}$ and $\\mathbf{Y}$ during deployment differs from the process that generated the training data. In this case, we can no longer expect the model's performance during deployment to match that observed on held out training data. If a *ground truth* is available during deployment, *supervised drift detection* can be performed, with the model's predictive performance monitored directly. However, in many deployment scenarios, such as the binary classification example below, a ground truth is not available and *unsupervised drift detection* methods are required.\n",
+    "\n",
+    "![drift in deployment](images/drift_deployment.png) \n",
+    "\n",
+    "To explore the different types of drift, consider the common scenario where we have input data $\\mathbf{X}$ and output data $\\mathbf{Y}$, jointly distributed according to $P(\\mathbf{X},\\mathbf{Y})$, and a model $f: x \\mapsto y$ trained on training/reference data with distribution $P_{ref}(\\mathbf{X},\\mathbf{Y})$. *Drift* is said to have occured when $P(\\mathbf{X},\\mathbf{Y}) \\ne P_{ref}(\\mathbf{X},\\mathbf{Y})$. Writing the joint distribution as\n",
```

Comment:
This is actually an important nuance: the training data should be disjoint from the reference data for a pretrained preprocessing step (eg encoder, model used for the uncertainty detectors etc).